### ISSUE TITLE
Direct3D 9: linear filtering applied too often

### ISSUE THREAD

**Issue by unknownbrackets** [hrydgard/ppsspp#7207]
I'm not sure if there is a solution to this.

A very clear example can be seen in Star Ocean, at the SQUARE ENIX logo.  This logo is a 4-bit clut texture, and uses transparent green as its background color.  In OpenGL, it works properly, but in Direct3D there's a green glow around it.

In this case, the texture is 256x256, and it's being drawn in through mode with coordinates exactly 256x256 apart.  So, the maginification/minification is 1:1.

The game, for whatever reason, sets the both min and mag to linear.  Changing mag to nearest solves the issue, so Direct3D seems to consider the texture as being magnified.  Somehow sounds similar to #6077, but in this case both min and mag are linear.

It's not using any testing whatsoever.  It is only using alpha blending to get rid of the green, because all green texture values are fully transparent.

-[Unknown]

**Comment by unknownbrackets** [comment]
I removed the offset from the shader manager (in `ConvertProjMatrixToD3DThrough()`) and simply did this before drawing the software primitives:

```c++
			for (int i = 0; i < maxIndex; ++i) {
				drawBuffer[i].x -= 0.5f;
				drawBuffer[i].y -= 0.5f;
			}
```

This "fixed" the issue.  Not sure if this means scaling via the matrix mul is losing too much precision or what.

-[Unknown]

**Comment by unknownbrackets** [comment]
This was fixed in #9717.

-[Unknown]
